### ISSUE TITLE
Changelog: document `Py_AtExit` requirement change that the interpreter must be initialized

### ISSUE THREAD

**Issue by mathstuf** [python/cpython#118016]
# Documentation

Python 3.12 has commit 2ad2bd8b14505ee92600e7988379cfac63086dab which now requires that the interpreter be initialized before calling `Py_AtExit`. Please add a retroactive note to the changelog for 3.12 and a new one to 3.13 (for those who don't go back to read the 3.12 changelogs) about this new change.

The API docs might also want to mention it too.

**Comment by ericsnowcurrently** [comment]
FWIW, the C-API docs indirectly indicate that users should not use `Py_AtExit()` before initialization:  https://docs.python.org/3.12/c-api/init.html#before-python-initialization.

(That doesn't mean an explicit note is a bad idea, of course.)

**Comment by mathstuf** [comment]
Indeed. It looks like we started using it in mid-2014 and it had worked by chance until 3.12 (with a segfault on the `NULL` mutex). So Hyrum's Law is definitely in effect here.
